shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -88,142 +151,106 @@ export function utlEnableStorage() {
  * @param reset - Should the usage be reset and determined only based on whether LocalStorage is available
  */
 export function utlCanUseLocalStorage(reset?: boolean): boolean {
-    if (reset || _canUseLocalStorage === undefined) {
-        _canUseLocalStorage = !!_getVerifiedStorageObject(StorageType.LocalStorage);
+    if (reset) {
+        _verifiedLocalStorage = null;
     }
-
-    return _canUseLocalStorage;
+    
+    return !!_getLocalStorageObject();
 }
 
 export function utlGetLocalStorage(logger: IDiagnosticLogger, name: string): string {
     const storage = _getLocalStorageObject();
     if (storage !== null) {
-        try {
-            return storage.getItem(name);
-        } catch (e) {
-            _canUseLocalStorage = false;
-
-            _throwInternal(logger,
-                eLoggingSeverity.WARNING,
-                _eInternalMessageId.BrowserCannotReadLocalStorage,
-                "Browser failed read of local storage. " + getExceptionName(e),
-                { exception: dumpObj(e) });
-        }
+        return storage.getItem(name);
```

Comment:
Refactor to have a single return and the smallest amount of code

Reply:
No you didn't revert to the original, keeping the _throwInternal, but no need to set the _canUseLocalStorage as this is now handled in the wrapped implementation.